clients/admin-ui/src/features/common/custom-fields/CustomFieldSelector.tsx

```diff
@@ -0,0 +1,65 @@
+import { useField } from "formik";
+import { useEffect } from "react";
+
+import { CustomSelect } from "~/features/common/form/inputs";
+import { AllowedTypes } from "~/types/api";
+
+import {
+  AllowListWithOptions,
+  CustomFieldDefinitionExisting,
+  CustomFieldExisting,
+} from "./types";
+
+type Props = {
+  allowList: AllowListWithOptions;
+  customField?: CustomFieldExisting;
+  customFieldDefinition: CustomFieldDefinitionExisting;
+};
+
+export const CustomFieldSelector = ({
+  allowList,
+  customField,
+  customFieldDefinition,
+}: Props) => {
+  const name = `definitionIdToCustomFieldValue.${customFieldDefinition.id}`;
+  const label = customFieldDefinition.name;
+  const tooltip = customFieldDefinition.description;
+  const { options } = allowList;
+
+  const [, meta, helpers] = useField({ name });
+
+  // This is a bit of a hack to set the initial value of the selector based on the CustomField
+  // returned by the API (if any). The "correct" way to do this would be to have the `initialValues`
+  // passed to `Formik` include the API values mapped by definition ID. However, because custom
+  // fields are so dynamic and depend on multiple API calls, mixing that logic into the (already
+  // complex) taxonomy form initialization isn't feasible for now. Those values are created by:
+  // src/features/taxonomy/hooks.tsx.
```

Comment:
Now that I've refactored the taxonomy hooks anyway, it's possible the initial values could get populated in the correct way. I couldn't figure out a good way to do that yet though.

Reply:
Nice work encapsulating this logic using additional hooks and into its own drop down component.

Reply:
Still don't think its possible to update the initial values in the CustomFieldsList component?

Reply:
No, I think it would have to be higher up, to where Formik actually receives the prop, which I think means it needs to fit into the taxonomy hooks' methods.